dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/meter/C3MeterChartDisplayer.java

```diff
@@ -0,0 +1,155 @@
+                   .supportsAttribute(DisplayerAttributeDef.RENDERER)
+                   .supportsAttribute(DisplayerAttributeGroupDef.COLUMNS_GROUP)
+                   .supportsAttribute(DisplayerAttributeGroupDef.FILTER_GROUP)
+                   .supportsAttribute(DisplayerAttributeGroupDef.REFRESH_GROUP)
+                   .supportsAttribute(DisplayerAttributeGroupDef.GENERAL_GROUP)
+                   .supportsAttribute(DisplayerAttributeDef.CHART_WIDTH)
+                   .supportsAttribute(DisplayerAttributeDef.CHART_HEIGHT)
+                   .supportsAttribute(DisplayerAttributeGroupDef.CHART_MARGIN_GROUP)
+                   .supportsAttribute(DisplayerAttributeGroupDef.CHART_LEGEND_GROUP)
+                   .supportsAttribute(DisplayerAttributeGroupDef.METER_GROUP);
+    }
+    
+    @Override
+    public View getView() {
+        return view;
+    }
+    
+    @Override
+    protected C3ChartConf buildConfiguration() {
+        C3ChartConf conf = super.buildConfiguration();
+        C3Gauge gaugeConf = createGauge();
+        conf.setGauge(gaugeConf);
+        return conf;
+    }
+    
+    @Override
+    protected String[][] createSeries() {
+        List<DataColumn> columns = dataSet.getColumns();
+        String[][] output = new String[0][0];
+        if (columns.size() == 1) {
+            output = new String[1][];
+            output[0] = extractSingleColumnValues(columns.get(0));
+        } else {
+            DataColumn groupsColumn = columns.get(0);
+            DataColumn valuesColumn = columns.get(1);
+            output = extractGroupingValues(groupsColumn, valuesColumn);
+        }
+        return output;
+    }
+    
+    @Override
+    protected C3Color createColor() {
+        C3Color color = super.createColor();
+        int[] thresholdValues = {
+                (int) displayerSettings.getMeterWarning(),
+                (int) displayerSettings.getMeterCritical(),
+                (int) displayerSettings.getMeterEnd()
+        };
+        C3Threshold c3Threshold = factory.c3Threshold(thresholdValues);
+        color.setPattern(getView().getColorPattern());
+        color.setThreshold(c3Threshold);
+        return color;
+    }
+    
+    private C3Gauge createGauge() {
+        int meterStart = (int) displayerSettings.getMeterStart();
+        int meterEnd = (int) displayerSettings.getMeterEnd();
+        return factory.c3Gauge(meterStart, meterEnd);
+    }
+
+    String[][] extractGroupingValues(DataColumn groupsColumn, DataColumn valuesColumn) {
+        int n = groupsColumn.getValues().size();
+        String[][] output = new String[n][2];
+        for (int i = 0; i < n; i++) {
+            Object group = groupsColumn.getValues().get(i);
+            Object value = valuesColumn.getValues().get(i);
+            String groupStr = "", valueStr = "";
+            if (group != null) {
+                groupStr = group.toString();
+            } else {
```

Comment:
Forgotten empty else here? Anyway I'd consider using ternary operator a bit more readable here:
`String groupStr = group == null ? "" : group.toString();`

Or even private helper that could be used 3x in this class.
```java
private String toString(String mightBeNull) {
  return mightBeNull == null ? "" : mightBeNull.toString();
}
```
But this is probably just a matter of taste. Leaving the decision up to you.

Reply:
@jesuino could you please take a look at this?